product.py

```diff
@@ -69,17 +69,26 @@ def elastic_search_json(self):
                 "true" if self.displayed_on_eshop else "false"
             ),
             'active': "true" if self.active else "false",
-            'attributes': self.get_elastic_filterable_data(),
+            'attributes': self.elastic_attributes_json(),
         }
 
-    def get_elastic_filterable_data(self):
+    def elastic_attributes_json(self):
         """
-        This method returns a dictionary of attributes which will be used to
-        filter search results. By default, it returns a product's attributes.
-        Downstream modules can override this method to add any other relevant
-        fields. This data is added to the index.
+        This method returns the filterable attributes of the product in a
+        <name>: <value> format for easy consumption by elasticsearch.
         """
-        return self.attributes
+        attr_names = map(lambda attr: attr.attribute.name, self.attributes)
+        attr_values = []
+
+        for attr in self.attributes:
```

Comment:
the if else logic here will be repeated in countless places, if not provided by core module.

Why not add a function field called value in product attributes, add it to the tree view as well and use the logic below as a getter ?